File: market-mapper/src/main/java/by/market/mapper/StringCharacteristicMapper.java
package by.market.mapper;

import by.market.domain.characteristics.single.StringCharacteristic;
import by.market.dto.characteristics.StringCharacteristicDTO;
import org.mapstruct.InheritInverseConfiguration;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(config = MapperConfiguration.class)
public interface StringCharacteristicMapper extends MapstructMapper<StringCharacteristicDTO, StringCharacteristic> {

    @Override
    @Mapping(source = "characteristic.id", target = "characteristic")
    @Mapping(source = "product.id", target = "product")
    StringCharacteristicDTO toMap(StringCharacteristic objectAbstractCharacteristic);

    @Override
    @InheritInverseConfiguration
    StringCharacteristic fromMap(StringCharacteristicDTO objectAbstractCharacteristicDTO);

}

File: market-mapper/src/main/kotlin/by/market/mapper/characteristics/single/StringCharacteristicMapper.kt
package by.market.mapper.characteristics.single

import by.market.domain.characteristics.single.StringCharacteristic
import by.market.dto.characteristics.StringCharacteristicDTO
import by.market.mapper.IMapstructMapper
import by.market.mapper.MapperConfig
import org.mapstruct.Mapper

@Mapper(config = MapperConfig::class)
interface StringCharacteristicMapper : IMapstructMapper<StringCharacteristicDTO, StringCharacteristic>
